bigquery_etl/operational_monitoring/__init__.py

```diff
@@ -0,0 +1,177 @@
+"""Generate and run the Operational Monitoring Queries."""
+import os
+import time
+import json
+import click
+import logging
+
+from datetime import datetime, date
+from pathlib import Path
+
+from google.cloud import storage, bigquery
+
+from bigquery_etl.util.common import render, write_sql
+
+
+QUERY_FILENAME = "query.sql"
+INIT_FILENAME = "init.sql"
+BUCKET_NAME = "operational_monitoring"
+PROJECTS_FOLDER = "projects/"
+OUTPUT_DIR = "sql/moz-fx-data-shared-prod/"
+PROD_PROJECT = "moz-fx-data-shared-prod"
+
+
+def _download_json_file(project, bucket, filename):
+    blob = bucket.get_blob(filename)
+    return json.loads(blob.download_as_string()), blob.updated
+
+
+def _get_name_and_sql(query, reference_content, item_type):
+    return [
+        {"name": item, "sql": reference_content[item]["sql"]}
+        for item in query[item_type]
+    ]
+
+
+def _write_sql(project, dataset, slug, kwargs, init):
+    basename = INIT_FILENAME if init else QUERY_FILENAME
+    write_sql(
+        OUTPUT_DIR,
+        f"{project}.{dataset}.{slug}",
+        basename,
+        render(
+            basename,
+            template_folder="operational_monitoring",
+            **kwargs,
+            init=init,
+        ),
+    )
+
+
+def _query_up_to_date(dataset, slug, basename, project_last_modified):
+    """Check whether the project file was updated after the query.sql file was written."""
+    query_path = Path(os.path.join(OUTPUT_DIR, dataset, slug, basename))
+    query_last_modified = datetime.utcfromtimestamp(query_path.stat().st_mtime)
+
+    if not os.path.exists(query_path) or (
+        os.path.exists(query_path)
+        and query_last_modified < project_last_modified.replace(tzinfo=None)
+    ):
+        return False
+
+    return True
+
+
+def _generate_sql(project, dataset):
+    render_kwargs = {
+        "header": "-- Generated via bigquery_etl.operational_monitoring\n",
+        "gcp_project": project,
+        "dataset": dataset,
+    }
+
+    client = storage.Client(project)
+    bucket = client.get_bucket(BUCKET_NAME)
+
+    probes, _ = _download_json_file(project, bucket, "probes.json")
+    dimensions, _ = _download_json_file(project, bucket, "dimensions.json")
+
+    # Iterating over all defined operational monitoring projects
+    for blob in bucket.list_blobs(prefix=PROJECTS_FOLDER):
```

Comment:
I haven't checked how long the generation takes, but if it takes significant time we might also run this in parallel